src/domains/shielder/stores/getShielderIndexedDB.ts

```diff
@@ -73,17 +83,35 @@ const initDB = async (): Promise<IDBPDatabase<DBSchema>> => {
   });
 };
 
+const initDB = async (): Promise<IDBPDatabase<DBSchema>> => {
+  try {
+    return await createDB();
+  } catch (error) {
+    if ((error as Error).name === 'VersionError') {
```

Comment:
`error as Error` is not type-safe. You should validate if error is an object (or just use zod for that entire `if`).